Sources/Recycling/ListViews/UserAds/Cell/UserAdsListViewCell.swift

```diff
@@ -0,0 +1,208 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+public enum UserAdStatus: String {
+    case started = "Påbegynt"
+    case active = "Aktiv"
+    case inactive = "Inaktiv"
+    case expired = "Utløpt"
+    case sold = "Solgt"
+    case unknown = "Ukjent"
+}
+
+public protocol UserAdsListViewCellDataSource: class {
+    func shouldDisplayCellAsInactive(_ userAdsListViewCell: UserAdsListViewCell) -> Bool
+    func userAdsListViewCell(_ userAdsListViewCell: UserAdsListViewCell, loadImageForModel model: UserAdsListViewModel, imageWidth: CGFloat, completion: @escaping ((UIImage?) -> Void))
+    func userAdsListViewCell(_ userAdsListViewCell: UserAdsListViewCell, cancelLoadingImageForModel model: UserAdsListViewModel, imageWidth: CGFloat)
+}
+
+public class UserAdsListViewCell: UITableViewCell {
+    // MARK: - External properties
+
+    /// The loading color is used to fill the image view while we load the image.
+
+    public var loadingColor: UIColor?
+
+    /// A data source for the loading of the image
+
+    public weak var dataSource: UserAdsListViewCellDataSource?
+
+    // MARK: - Internal properties
+
+    private static let cornerRadius: CGFloat = 12
+    private static let imageSize: CGFloat = 100
+
+    private static let inactiveCornerRadius: CGFloat = 12
+    private static let inactiveImageSize: CGFloat = 56
+
+    private var defaultImage: UIImage? {
+        return UIImage(named: .noImage)
+    }
+
+    private lazy var adImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.layer.cornerRadius = UserAdsListViewCell.cornerRadius
+        imageView.layer.masksToBounds = true
+        imageView.contentMode = .scaleAspectFill
+        return imageView
+    }()
+
+    private lazy var titleLabel: Label = {
+        let label = Label(style: .title4)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.backgroundColor = .clear
+        return label
+    }()
+
+    private lazy var priceLabel: Label = {
+        let label = Label(style: .title5)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.backgroundColor = .clear
+        return label
+    }()
+
+    private lazy var detailLabel: Label = {
+        let label = Label(style: .detail)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = .stone
+        label.backgroundColor = .clear
+        return label
+    }()
+
+    private lazy var userAdStatus: UserAdStatus = .unknown
+
+    // MARK: - Setup
+
+    private func setupView() {
+        isAccessibilityElement = true
+        backgroundColor = .milk
+        accessoryType = .disclosureIndicator
+        selectionStyle = .none
+
+        let ribbonView = createRibbonView(forUserAdStatus: userAdStatus)
+
+        addSubview(ribbonView)
```

Comment:
Should you perhaps remove all the subviews at first seeing that `setupView` could potentially be called multiple times? I don't forget about `ribbonView`, should probably be an optional instance var then.

Reply:
Good point. Can call a teardown function in the `didSet` of the model before doing any setup/initialisation.